src/builder.py

```diff
@@ -276,8 +279,26 @@ def get_new_executable(self):
         return path
 
 
-class HostBuilder:
-    def __init__(self, build_directory, install_directory):
+class Abstract(NotImplementedError):
+    pass  # TODO: use abc?
+
+
+class Builder:
+    def __init__(self, build_directory: BuildDirectory,
+                 install_directory: Optional[str]) -> None:
+        self._build_directory = build_directory
+        self._install_directory = install_directory
+
+    def cmake(self, builddir: str, srcdir: str,
+              args: List[str]) -> None: raise Abstract()
+
+    def install(self, source: str, target: str) -> None: raise Abstract()
+
+    def run(self, cwd: str, command: List[str]) -> None: raise Abstract()
```

Comment:
Instead of raising `Abstract`, we can probably use `@abstractmethod` here.

Reply:
yes...